src/Terminal.ts

```
@@ -703,7 +705,11 @@ export class Terminal extends EventEmitter implements ITerminal, IDisposable, II
     // Performance: Add viewport and helper elements from the fragment
     this.element.appendChild(fragment);
 
-    this.renderer = new Renderer(this, this.options.theme);
+    if (this.options.rendererType === 'canvas') {
```

The only known renderer type seems to be 'canvas', any other string will use the dom renderer. Shouldn't we expect that 'dom' is passed if the DOM renderer should be used, and throw an exception if an unsupported renderer name gets passed?

---

👍 will fix this up